### ISSUE TITLE
Feature request: Render configuration option for GOP size

### ISSUE THREAD

**Issue by atticoos** [remotion-dev/remotion#4002]
# Feature Request 🛍️

Allow configuration the GOP size of the output video.

Add `gopSize: number` as an option to:
- `@remotion/renderer`:`renderMedia`
- `@remotion/lambda/client`: `renderMediaOnLambda`

## Use Case

As Remotion becomes used in more places and its output integrated with more services -- making additional encoding options available will be helpful to avoid having to re encode the output video to achieve certain video requirements.

We use [Mux](https://www.mux.com/) for video delivery, and they have a set of [standard input properties](https://docs.mux.com/guides/minimize-processing-time#standard-input-specs) for video to avoid making a full re encode. One requirement is a maximum GOP size of `239`

We've been able to satisfy the request of the requirements, such as `yup420p` pixel format, and specific bitrates with Remotion's existing configuration options -- would be great to configure GOP size as well for folks that need to get precise.

**Comment by JonnyBurger** [comment]
Good idea, I am just not familiar with it. Which FFmpeg flags would this expose exactly? What should be the defaults? Which codecs out of the ones that we support support this parameter as well?

**Comment by atticoos** [comment]
This is ffmpeg's `-g` flag, and by default, can be omitted unless otherwise specified by a value. While all video codecs that use I/P/B frames have GOPs, different codecs may prefer different GOP configurations but generally have the same implications

**Comment by JonnyBurger** [comment]
This was implemented in https://github.com/remotion-dev/remotion/pull/7610!
